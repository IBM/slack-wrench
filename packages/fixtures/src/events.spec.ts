import { events } from './index';

describe('Events fixtures', () => {
  const user_id = 'UPINKIE';
  const text = 'Rainbow is 20% cooler';

  it('generates a message block', () => {
    expect.assertions(1);

    // Not including more in depth tests as typing should serve that purpose
    expect(events.message(text).event).toEqual(
      expect.objectContaining({ text }),
    );
  });

  it('can override message fields', () => {
    expect.assertions(1);

    const options = { user: user_id };

    expect(events.message(text, options).event).toEqual(
      expect.objectContaining(options),
    );
  });

  it('generates a appMention block', () => {
    expect.assertions(1);

    // Not including more in depth tests as typing should serve that purpose
    expect(events.appMention(text).event).toEqual(
      expect.objectContaining({ text }),
    );
  });

  it('can override appMention fields', () => {
    expect.assertions(1);

    const options = { user: user_id };

    expect(events.appMention(text, options).event).toEqual(
      expect.objectContaining(options),
    );
  });

  it('generates a memberJoinedChannel block', () => {
    expect.assertions(1);

    expect(events.memberJoinedChannel()).toMatchSnapshot();
  });

  it('can override memberJoinedChannel fields', () => {
    expect.assertions(1);

    const options = {
      user: 'UNEW_USER',
      channel: 'CNEW_CHANNEL',
      team: 'TNEW_TEAM',
      channel_type: 'G',
    };
    expect(events.memberJoinedChannel(options).event).toEqual(
      expect.objectContaining(options),
    );
  });

  it('channel type is inferred in memberJoinedChannel event', () => {
    expect.assertions(1);

    const options = { channel: 'GNEW_CHANNEL' };
    const result = { channel_type: 'G' };
    expect(events.memberJoinedChannel(options).event).toEqual(
      expect.objectContaining(result),
    );
  });
});
