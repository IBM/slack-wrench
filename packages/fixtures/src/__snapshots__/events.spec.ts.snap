// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events fixtures generates a memberJoinedChannel block 1`] = `
Object {
  "api_app_id": "API_APP_ID",
  "authed_users": Array [
    "UUSERID",
  ],
  "event": Object {
    "channel": "CCHANNELID",
    "channel_type": "C",
    "inviter": "IINVITERID",
    "team": "TTEAMID",
    "type": "member_joined_channel",
    "user": "UUSERID",
  },
  "event_id": "EVENT_ID",
  "event_time": 1234567890,
  "team_id": "TTEAMID",
  "token": "TOKEN",
  "type": "event_callback",
}
`;
